### ISSUE TITLE
Translations

### ISSUE THREAD

**Issue by alensiljak** [orgzly-revived/orgzly-android-revived#46]
Let's work out the Translations here.

Right now, we link to the Orgzly translations in the Readme doc. Since I assume we can't maintain those, either, we probably have to come up with a solution for people to continue translating the app.

I'm not sure if we can get the translations from the existing Orgzly project, wherever it is hosted.

Ideas?

**Comment by cbkerr** [comment]
Currently, the translation effort is hosted at [Crowdin](https://crowdin.com/project/orgzly). It is still active, with at least 7 people suggesting roughly 150 translations in the last year. I didn't find any documentation about transferring ownership of a project to a different organization without the owner present.

I made an account and poked around. It looks like I am able to download the translated files.

Since the history is kept for 1 year and the [last manual update was Nov 5 2022](https://github.com/orgzly-revived/orgzly-android-revived/commit/70c81822464437935d25ebe616ef125fc48077a6), I can check the [activity](https://crowdin.com/project/orgzly/activity-stream) and download the files with updates. Then re-upload to GitHub.

In the future, it would be good to have a more automatic way of doing this, like [integration with GitHub](https://support.crowdin.com/github-integration/).

If we want to change providers, I think a good option is [Weblate](https://weblate.org/en/), which is libre software and has a hosted version for foss projects with a limit of 10,000 strings. We only have <800 so far. Here is an example discussion comparing it to some other options: https://github.com/mumble-voip/mumble/discussions/4641?sort=new

I am not sure if the current translators would move with the project if we changed providers, or if we would attract more volunteers hosted at a new place.

**Comment by alensiljak** [comment]
Just a quick note - by staying at Crowdin, all the existing contributors could just switch projects. The project could reuse existing translations (provided as suggestions). 
Also, it should be easier to merge contributions in case people continue to contribute to Orgzly project there.

Not that I have preferences to any of the sites. I've used Crowdin in the past for another OS project and was doing all the work of adding the translations into the app manually. I guess the same is happening with Orgzly at the moment.

**Comment by inson1** [comment]
I would be for Hosted Weblate. Its open source.

**Comment by amberin** [comment]
Yes, let's create Orgzly Revived as a Crowdin project. What do we need to do that?

**Comment by amberin** [comment]
I couldn't figure out how to clone/copy the project, and I suspect it's not possible without the owner being involved. But I have now created a new project and synced it with our repo: https://crowdin.com/project/orgzly-revived

**Comment by amberin** [comment]
What's frustrating is that I can see that some heroes have made contributions to the original Crowdin project quite recently, and I don't think we'll be able to import those.

**Comment by amberin** [comment]
The Crowdin project seems to work fine with a Github integration, but my account is currently in some free trial period and it's unclear to me if we'll lose some important functionality when the trial period ends. Let's keep this issue open until then.

**Comment by amberin** [comment]
Yeah, the Crowdin integration is not working since we fell back to the Free plan. I'll ask the support if any Github integration at all is possible for free.

**Comment by amberin** [comment]
I have added a Github Action which runs daily and syncs translations with Crowdin.

It pushes a commit to a branch named crowdin something. It will not open a PR, because that requires opening up permissions for all Actions. I'm thinking it's easy to create and merge a PR before releasing a new version, if the crowdin branch exists in the repo (as this means new translations have been downloaded).

Only files which are 100 % translated will be downloaded from Crowdin. I tried turning this limitation off, but then I noticed that it looked like some languages were losing translations which were present in the repo, but had not been entered in Crowdin. So I feel it's safer to only download fully translated files.

**Comment by amberin** [comment]
I agree that the 100% limitation feels a bit strict... I will make a new sync attempt with it turned off, and take a closer look.

**Comment by amberin** [comment]
The problem when disabling the 100% limitation is that some existing translations in the "Getting started"-notebook are not being recognized when imported into Crowdin. So with the Czech, Hungarian, Lithuanian and Ukrainian files, Crowdin considers them to be 0 % translated, and exports the original English strings. If we merge that export, we lose all the existing translations. I'm thinking people with knowledge of those languages could probably quite easily re-enter the existing translations, reaching 100% status in Crowdin. (Also, in most cases, I believe Crowdin will suggest the right translation from its "global TM".)

**Comment by alensiljak** [comment]
Hm, that's sad. The problem on the other end is, if a new string is added, then pretty much all translations will be below 100%. That means they won't pick up any additions to translation that may have occur, i.e between 60% and 80%.

**Comment by amberin** [comment]
Yes. I also wonder why most of the "getting started"-notebook files are imported without issues, while some are at 0%. Could it be that the problematic ones are too old, and therefore missing too much content?

**Comment by amberin** [comment]
@nevenz wrote:

> if a new string is added, then pretty much all translations will be below 100%.

Only the file where the string is added. And if it was previously at 100%, it just means the file won't be downloaded until that new string is translated - quite expected.

**Comment by alensiljak** [comment]
> [@nevenz](https://github.com/nevenz) wrote:
> 
> > if a new string is added, then pretty much all translations will be below 100%.
> 
> Only the file where the string is added. And if it was previously at 100%, it just means the file won't be downloaded until that new string is translated, which makes sense.

In simple cases that is true. But it also won't pick up any other potential modifications in the meantime.

**Comment by amberin** [comment]
> Hello,

> As a result of a recent source files update, you’ve exceeded the hosted words limit for your Crowdin account (username: amberin). Please upgrade your plan to keep making more сontent multilingual. Otherwise, your account will be suspended within 10 days.

The cheapest paid plan in Crowdin is 48 EUR/month. That's a lot. I suggest we look at migrating to Weblate.

**Comment by amberin** [comment]
@yurtpage That sounds great - let's try that! Thank you!

**Comment by alensiljak** [comment]
With the progress of AI, I wonder how long until all such tools are completely irrelevant.
With Ollama running locally the gemma3 (multilingual) model, and the use of MCPs, I think it would be possible to create something that would translate the sources into any number of languages.

**Comment by amberin** [comment]
That's true for translations in general, but I'd wager the lack of context will render very poor results. (This is of course a challenge in e.g. Crowdin for humans as well, but I think the value is in human users being able to improve poor translations themselves.)

**Comment by amberin** [comment]
@yurtpage Thank you, I have gained admin rights in Toolate and I think it looks very good, at first glance. I also noticed that Toolate is aware of the Fastlane string length rules, which will help us woth #547. That's good news.

**Comment by IzzySoft** [comment]
Looks like it's too late to jump in, but Toolate returns 404 for the translation links from your Readme?

**Comment by amberin** [comment]
Yup. Hopefully temporary; there is a notice about ongoing server migration at https://toolate.othing.xyz/.

**Comment by IzzySoft** [comment]
Thanks! Fingers crossed then… 🤞
